projects/lib/src/lib/remote-data.ts

```diff
@@ -26,13 +26,24 @@ export class InProgress<T> {
   }
 }
 
-export class Failure<E> {
+export class Failure<E, T> {
   private tag = RemoteDataTags.Failure;
-  private constructor(private err: E) {}
-  static of<T, E = DefaultError>(err: E): RemoteData<T, E> {
-    return new Failure(err);
+  private constructor(private err: E, private val?: T) {}
+  static of<T, E = DefaultError>(err: E, val?: T): RemoteData<T, E> {
+    if (val) {
```

Comment:
This if/else statement is redundant, `new Failure(err, val);` does exactly the same